howtos/diffs/lr-schedule.diff

```diff
@@ -0,0 +1,87 @@
+diff --git a/examples/mnist/train.py b/examples/mnist/train.py
+index 9c21985..41925dc 100644
+--- a/examples/mnist/train.py
++++ b/examples/mnist/train.py
+@@ -84,6 +84,18 @@ def create_model(key):
+   return model
+ 
+ 
++def create_triangular_schedule(lr_min, lr_max, epochs):
++  """Return fn from epoch to LR, linearly interpolating between `lr_min`->`lr_max`->`lr_min`."""
++  top = epochs // 2
++  def learning_rate_fn(epoch):
++    if epoch < top:
++      lr = lr_min + epoch/top * (lr_max - lr_min)
++    else:
++      lr = lr_max - ((epoch - top)/top) * (lr_max - lr_min)
```

Comment:
Minor nit: LR schedule could wind up negative here. Wasn't sure if that was intended behavior. Marking as a "changes requested", but mostly just want to make sure there's a brief discussion

Reply:
Good point, I'm against that. So if I round up top, we get `epoch <= 2*top`, which leads to the minimum value of `lr` being `lr_min` and max being `lr_max`